x-pack/plugins/file_upload/public/importer/geo/shapefile_importer/side_car_file_picker.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Component }  from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiFilePicker, EuiFormRow } from '@elastic/eui';
+
+function getFileNameWithoutExt(fileName: string) {
+  const split = fileName.split('.');
+  return split.length ? split[0] : fileName;
```

Comment:
```suggestion
  const extPos= fileName.lastIndexOf('.');
  return extPos !== -1 ? fileName.substring(0, extPos) : fileName;
```

Nit: I think something like this would avoid a case where a shapefile like `foo.bar.shp` and a sidecar like `foo.buzz.dbf` get matched erroneously since it's possible for shapefile names to contain periods.